### PR TITLE
[IMP] l10n_ch: Gen QR-bill and ISR only for CH&LI

### PR DESCRIPTION
**Pull request by yvaucher**
**Description of the issue/feature this PR addresses:**

Multi-company context

The main issue is with the QR-bill as it's basic condition to generate
it, is to have an IBAN account. Which most countries will have.

This removes anoyances from l10n_ch on non Swiss and Liechtenstein
companies.

**Current behavior before PR:**

In multi company, when `l10n_ch` is installed, QR-bill report is attached to each and every invoice emails when doing `action_invoice_sent`. (the only condition is that a bank account is set and it has an IBAN, which virtually means all invoices)

**Desired behavior after PR is merged:**

Only Switzerland and Liechtenstein use QR-bill and ISR reports

For any other countries, never attach QR-bill in invoice emails.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/l10n_ch/models/account_invoice.py [hunk 1]
```diff
@@ -233,19 +233,21 @@ def _compute_l10n_ch_isr_optical_line(self):
...
                 record.l10n_ch_isr_subscription and \
-                record.l10n_ch_currency_name in ['EUR', 'CHF']
+                record.l10n_ch_currency_name in ['EUR', 'CHF'] and \
+                record.company_id.country_id.code in ["CH", "LI"]
 
     @api.depends('type', 'invoice_partner_bank_id', 'invoice_payment_ref')
     def _compute_l10n_ch_isr_needs_fixing(self):
         for inv in self:
-            if inv.type == 'in_invoice' and inv.company_id.country_id.code == "CH":
+            if inv.type == 'in_invoice' and inv.company_id.country_id.code in ["CH", "LI"]:
```
Comment (jco-odoo) [inline addons/l10n_ch/models/account_invoice.py:250]: @yvaucher Here it makes sense, but I don't see immediately the added value in v13 for the other cases where you add it. Does LI also need cash rounding, ...  Are they happy enough using l10n_ch?

Comment (yvaucher) [reply addons/l10n_ch/models/account_invoice.py:250]: I have no idea, I just know they use QR-bill. And if they use QR-bill they might need the reference type ISR.

Here the aim is mostly to not have the QR-bill attached to France, Germany or Belgium (for example) companies when installed on the same database as a company in Switzerland. It messes up the mail sent.

So I reduced
* _all countries of the world_
to
* _the 2 countries that uses QR-bill_

Then updated other country code check to be consistent. (but the one in deprecated function)

I don't even know if there are user in Liechtenstein. (I hope there are :) )

From [Swiss Implementation Guidelines forthe QR-bill](https://www.paymentstandards.ch/dam/downloads/ig-qr-bill-en.pdf)

> It should also be noted that the QR-bill is based on the ISO20022 standard and was designed primarily for invoicing in Switzerland and Liechtenstein. [...]

IMO, it's another reason why the QR-bill should not be part of `l10n_ch` which should only include the CoA.

**Comment by yvaucher** [comment]
@jco-odoo should I remove the Lichtenstein country code to get this patch merged?

**Comment by jco-odoo** [comment]
@yvaucher If you say that it works more or less with Liechtenstein, I don't mind.  But if you get a vendor bill from a Swiss person and you are in France, you should not be able to pay through the ISR reference system?  

And otherwise for the customer invoices it would make sense as it would be weird to have a main Swiss bank account?

**Comment by yvaucher** [comment]
For Liechtenstein, it should work in principle according to the QR-Bill spec. I do not have any customer that lives in Liechtenstein to confirm it thought.

**If you are in France and receive a bill from a Swiss person:**

This PR doesn't impact this case as it fixes emitting invoices for non-swiss company.

:switzerland:  company: customer invoice mail sent with QR-Bill :heavy_check_mark: 
:fr:  company: customer invoice mail sent with QR-Bill :negative_squared_cross_mark:  -> This PR fix this by removing the QR-Bill from emitted invoices.

We want the French (or any non CH or LI) company to emit invoice mails without a QR-Bill due to a sister company in an other country.

The only case I see is a Company based in France :fr:  wants to emit invoices to Switzerland :switzerland:  customers. Thus to use QR this company could have Swiss bank account. But, then this company wouldn't want to install the whole `l10n_ch` module as it's not looking to install the Swiss CoA. Thus, in the current state of the module I suggest to disregard this specific case.

**Comment by C3POdoo** [comment]
Dear @yvaucher,

Thank you for your contribution but the version 13.0 is no longer supported.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense for the upper version, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**